packages/block-library/src/navigation-menu/block-navigator.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Fragment,
+	useState,
+} from '@wordpress/element';
+import {
+	useSelect,
+	useDispatch,
+} from '@wordpress/data';
+import {
+	BlockNavigationList,
+} from '@wordpress/block-editor';
+import {
+	IconButton,
+	SVG,
+	Path,
+	Modal,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const NavigatorIcon = (
+	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="20" height="20">
+		<Path d="M5 5H3v2h2V5zm3 8h11v-2H8v2zm9-8H6v2h11V5zM7 11H5v2h2v-2zm0 8h2v-2H7v2zm3-2v2h11v-2H10z" />
+	</SVG>
+);
+
+export default function BlockNavigator( { clientId } ) {
+	const [ isNavigationListOpen, setIsNavigationListOpen ] = useState( false );
+
+	const {
+		block,
+		selectedBlockClientId,
+	} = useSelect( ( select ) => {
+		const {
+			getSelectedBlockClientId,
+			getBlock,
+		} = select( 'core/block-editor' );
+
+		return {
+			block: getBlock( clientId ),
+			selectedBlockClientId: getSelectedBlockClientId(),
+		};
+	}, [ clientId ] );
+
+	const {
+		selectBlock,
+	} = useDispatch( 'core/block-editor' );
+
+	return (
+		<Fragment>
+			{ isNavigationListOpen && (
+				<Modal
```

Comment:
I would have expected that the Navigation block would use both modal and the Block navigator to make up the UX in such way. Is this better to have the `block navigator` _be_ a modal?

Reply:
Yeah, it might be good to not have the modal render as a child of the block toolbar, considering how the toolbar likes to unmount quite a lot.

Also keen to not add too much code to the main Navigation Menu edit component though or make this any more complicated than it needs to be.

Reply:
I've switched to using a hook for the block navigator and I'm now rendering the modal outside of the toolbar. It probably won't stay this way. I could imagine that if the navigator is something we'd consider using for other blocks, then the modal would be moved to the editor itself and be opened using an action.

This is a pragmatic solution for now.